python/ray/dag/compiled_dag_node.py

```diff
@@ -709,8 +709,17 @@ def _get_or_compile(
                 # in the DAG.
                 readers = [self.idx_to_task[idx] for idx in task.downstream_node_idxs]
 
-                if isinstance(readers[0].dag_node, MultiOutputNode):
-                    assert len(readers) == 1
+                dag_nodes = [reader.dag_node for reader in readers]
+                has_multi_output_node = False
```

Comment:
```suggestion
                read_by_driver = False
```

I think this is a bit more clear.

Reply:
Done